src/main/java/net/mcreator/ui/init/ArmorMakerTexturesCache.java

```diff
@@ -20,63 +20,48 @@
 
 import net.mcreator.io.ResourcePointer;
 import net.mcreator.io.TemplatesLoader;
-import net.mcreator.io.UserFolderManager;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.jetbrains.annotations.Nullable;
 
 import javax.imageio.ImageIO;
 import javax.swing.*;
-import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 
 public class ArmorMakerTexturesCache {
 	private static final Logger LOG = LogManager.getLogger("Armor Texture Cache");
-	private static final String MCR_FOLDER = System.getProperty("user.home")+ "\\.mcreator\\";
+	private static final String MCR_FOLDER = System.getProperty("user.home") + "\\.mcreator\\";
 
-	public static final Map<String, ImageIcon> CACHE = new ConcurrentHashMap<>();
-	public static final List<String> NAMES = new ArrayList<>();
+	private static final Map<String, ImageIcon> CACHE = new ConcurrentHashMap<>();
 
 	public static void init() {
 		List<ResourcePointer> templatesSorted = TemplatesLoader.loadTemplates("textures.armormaker", "png");
 		ImageIO.setUseCache(false); // we use custom image cache for this
 		templatesSorted.forEach(resourcePointer -> {
 			try {
-				CACHE.put(resourcePointer.identifier.toString(), new ImageIcon(ImageIO.read(resourcePointer.getStream())));
+				CACHE.put(resourcePointer.identifier.toString(),
+						new ImageIcon(ImageIO.read(resourcePointer.getStream())));
 			} catch (Exception e) {
 				LOG.warn("Failed to load armor texture from template: " + resourcePointer.identifier);
 			}
 		});
 		ImageIO.setUseCache(true);
-

```

Comment:
I think it was ok to cache this part too. Any reason you moved this out into armor maker?

Reply:
I thought it was what you meant.

Reply:
I meant there was no need for a separate NAMES field, as you could obtain them from the Map. But looking now, you couldn't as the lists differ. It was ok before, sorry for this confusion.

But consider making it Set instead of List if you don't want duplicate values, so you don't need to do contains check.